src/server/req_delete.c

```diff
@@ -362,230 +390,288 @@ req_deletejob(struct batch_request *preq)
 	int rc = 0;
 	int delhist = 0;
 	int err = PBSE_NONE;
-
-	snprintf(jid, sizeof(jid), "%s", preq->rq_ind.rq_delete.rq_objname);
-
-	if (preq->rq_extend && strstr(preq->rq_extend, DELETEHISTORY))
-		delhist = 1;
-	if (preq->rq_extend && strstr(preq->rq_extend, FORCE))
-		forcedel = 1;
-	/* with nomail , nomail_force , nomail_deletehist or nomailforce_deletehist options are set
-	 *  no mail is sent
-	 */
-	if (preq->rq_extend && strstr(preq->rq_extend, NOMAIL))
-		qdel_mail = 0;
-	else
-		qdel_mail = 1;
-
-	parent = chk_job_request(jid, preq, &jt, &err);
-	if (parent == NULL) {
-		pjob = find_job(jid);
-		if (pjob != NULL && pjob->ji_pmt_preq != NULL)
-			reply_preempt_jobs_request(err, PREEMPT_METHOD_DELETE, pjob);
-		return; /* note, req_reject already called */
-	}
-
-	if (delhist) {
-		rc = check_deletehistoryjob(preq);
-		if (rc == TRUE)
-			return;
+	char **list;
+	int count;
+	int j;
+	struct batch_reply *preply = &preq->rq_reply;
+	CLEAR_HEAD(preply->brp_un.brp_delstat);
+	preply->brp_count = 0;
+	
+	if (preq->rq_type == PBS_BATCH_DeleteJobList) {
+		preply->brp_choice = BATCH_REPLY_CHOICE_Delete;
+		list = preq->rq_ind.rq_delete.rq_jobslist;
+		count = preq->rq_ind.rq_delete.rq_count;
+	} else {
+		list = break_comma_list(preq->rq_ind.rq_delete.rq_objname);
+		count = 1;
 	}
-
-
-	if (jt == IS_ARRAY_NO) {
-
-		/* just a regular job, pass it on down the line and be done
-		 * If the request is to purge the history of the job then set ji_deletehistory to 1
+	
+	preq->rq_ind.rq_delete.tot_jobs = count;
+	
+	for (j = 0; list[j] != NULL; j++) {
+		
+		snprintf(jid, sizeof(jid), "%s", list[j]);
+		
+		if (preq->rq_extend && strstr(preq->rq_extend, DELETEHISTORY))
```

Comment:
move this (and below all where check extend and decided what is requested) code outside of for loop